pandas/tests/arrays/sparse/test_array.py

```diff
@@ -266,6 +266,13 @@ def test_getitem_bool_sparse_array(self):
         exp = SparseArray([np.nan, 3, 5])
         tm.assert_sp_array_equal(res, exp)
 
+    # GH 45110
+    def test_getitem_bool_sparse_array_as_comparison(self):
+        arr = SparseArray([1, 2, 3, 4, np.nan, np.nan], fill_value=np.nan)
+        res = arr[arr > 2]
```

Comment:
I think you mentioned, that we don't have tests here? Could you add one with boolean indexer too?

Reply:
No at all, I meant that we don't have test for comparisons SA. Actually we have, but it test for Series (when SparseArray casts to Series), now I've changed it to version without cast that means correct.

For simple boolean mask I added test in my previous PR that was merged. You can see it above test that marked here by you.